src/app.js

```diff
@@ -1 +1,44 @@
+/* eslint-disable no-shadow */
+/* eslint-disable no-console */
 'use strict';
+
+const fs = require('fs');
+
+const copyFile = (source, destination) => {
+  if (!source || !destination) {
+    throw new Error('Source and destination must be provided');
+  }
+
+  if (source === destination) {
+    throw new Error('Source and destination paths are the same');
+  }
+
+  if (!fs.existsSync(source)) {
+    throw new Error(`${source} doesn't exist`);
+  }
+
+  const sourceStats = fs.statSync(source);
+
+  if (sourceStats.isDirectory()) {
+    throw new Error('Source is a directory, not a file');
```

Comment:
The error message could specify that the operation is intended for files, not directories.